core-data/resolvers.js

```diff
@@ -33,16 +43,37 @@ export async function* getAuthors() {
 	yield receiveUserQuery( 'authors', users );
 }
 
+async function* loadKindEntities( state, kind ) {
+	const hasEntities = hasEntitiesByKind( state, kind );
+
+	if ( hasEntities ) {
+		return;
+	}
+
+	const kindConfig = find( kinds, { name: kind } );
+	if ( ! kindConfig ) {
+		return;
+	}
+
+	const entities = await kindConfig.loadEntities();
+	yield addEntities( entities );
+}
+
 /**
- * Requests a entity's record from the REST API.
+ * Requests an entity's record from the REST API.
  *
  * @param {Object} state  State tree
  * @param {string} kind   Entity kind.
  * @param {string} name   Entity name.
  * @param {number} key    Record's key
  */
 export async function* getEntityRecord( state, kind, name, key ) {
-	const entity = getEntity( kind, name );
+	yield* loadKindEntities( state, kind );
+	// I can't use the state because it's outdated at this point
```

Comment:
Could we use the yielded result from `loadKindEntities`?

Reply:
The problem is `loadKindEntities` don't do anything if the entities are loaded. It can be updated to return the entities in all cases. So it can be renamed `getKindEntities` which could be just a regular resolver for the `getKindEntities` selector and that's the point where I felt the need of a `resolve` API.

`yield resolve( 'core' ).getKindEntities()` or something like that where I won't be obliged to call the selector `getKindEntities` by myself inside the resolver but it will be called automatically when the resolver finishes. 

I guess it's still not clear if it will be a common use-case etc... I prefer to wait a bit with and keep the current "shortcut"

Reply:
Ok so to simplify unit-testing, I went ahead and returned the available entities in all use-cases.